src/d/actor/d_a_tpota.cpp

```diff
@@ -5,71 +5,171 @@
 
 #include "d/actor/d_a_tpota.h"
 #include "d/d_procname.h"
+#include "JSystem/JGeometry.h"
 
+#include "JSystem/JParticle/JPAParticle.h"
+#include "d/d_particle.h"
+#include "d/d_com_inf_game.h"
+
+
+const u16 l_daTpota_idx_table[2] = {0x82AE, 0x82AF};
 /* 00000078-000001D0       .text _create__9daTpota_cFv */
 cPhs_State daTpota_c::_create() {
-    /* Nonmatching */
+    fopAcM_SetupActor(this, daTpota_c);
+    int count = 0;
+    JPABaseEmitter* emitter;
+
+    do {
+        position[count] = current.pos;
+        angle[count] = current.angle;
+    
+        emitter = dComIfGp_particle_set(
+            l_daTpota_idx_table[count],
+            &position[count], 
+            &angle[count], 
+            NULL, 0xff, NULL, -1, NULL, NULL, NULL
+        );
+        emitters[count] = emitter;
+        count++;
+    }while (count < 2);
+    return 4;
 }
 
 /* 00000250-00000298       .text _delete__9daTpota_cFv */
 bool daTpota_c::_delete() {
-    /* Nonmatching */
+    int count = 0;
+    while (count < 2) {
+        if (emitters[count] != NULL) {
+            emitters[count]->becomeInvalidEmitter();
+            emitters[count] = 0;
+        }
+        count++;
+    }
+    return 1;
 }
 
 /* 00000298-000002FC       .text make_ripple__9daTpota_cF4cXyz */
-void daTpota_c::make_ripple(cXyz) {
-    /* Nonmatching */
+void daTpota_c::make_ripple(cXyz i_position) {
+    dComIfGp_particle_set(dPa_name::ID_SCENE_82B0, &i_position,
+        NULL, NULL, 0xff, NULL, -1, NULL, NULL, NULL);
+    return;
 }
 
+
+
 /* 000002FC-00000354       .text check_water_h__9daTpota_cFP15JPABaseParticlef */
-void daTpota_c::check_water_h(JPABaseParticle*, float) {
-    /* Nonmatching */
+int daTpota_c::check_water_h(JPABaseParticle* i_ptcl, float i_position_y) {
```

Comment:
```suggestion
BOOL daTpota_c::check_water_h(JPABaseParticle* i_ptcl, float i_position_y) {
```
This is probably a boolean, so you can change the return type, and also ret = FALSE/TRUE